src/librustdoc/html/render.rs

```diff
@@ -879,8 +901,8 @@ themePicker.onclick = function() {{
     }
 
     fn show_item(item: &IndexItem, krate: &str) -> String {
-        format!("{{'crate':'{}','ty':{},'name':'{}','path':'{}'{}}}",
-                krate, item.ty as usize, item.name, item.path,
+        format!("{{'crate':'{}','ty':{},'name':'{}','desc':'{}','p':'{}'{}}}",
+                krate, item.ty as usize, item.name, item.desc.replace("'", "\\'"), item.path,
```

Comment:
`.replace("'", "\\'")` isn't how you escape strings for javascript. You need to use something like `as_json`.

Reply:
Bad surprise, quotes aren't escaped when using `Json` object or json conversion functions so for now, it's the best solution.